src/System.Management.Automation/engine/PropertyAccessor.cs

```diff
@@ -145,41 +145,24 @@ internal JsonConfigFileAccessor()
 
             //
             // Sets the per-user configuration directory
+            // Note: This directory may or may not exist depending upon the 
+            // execution scenario. Writes will attempt to create the directory
+            // if it does not already exist.
             //
             appDataConfigDirectory = Utils.GetUserConfigurationDirectory();
-            if (!Directory.Exists(appDataConfigDirectory))
-            {
-                try
-                {
-                    Directory.CreateDirectory(appDataConfigDirectory);
-                }
-                catch (UnauthorizedAccessException)
-                {
-                    // Do nothing now. This failure shouldn't block initialization
-                    appDataConfigDirectory = null;
-                }
-            }
         }
 
         /// <summary>
-        /// Enables delayed creation of the user settings directory so it does 
-        /// not interfere with PowerShell initialization
+        /// Attempts to create the user's configuration settings directory if it does not already exist.
         /// </summary>
-        /// <returns>Returns the directory if present or creatable. Throws otherwise.</returns>
-        private string GetCurrentUserConfigDirectory()
+        /// <returns></returns>
+        private void CreateUserConfigDirectory()
         {
-            if (null == appDataConfigDirectory)
+            if (!Directory.Exists(appDataConfigDirectory))
             {
-                string tempAppDataConfigDir = Utils.GetUserConfigurationDirectory();
-                if (!Directory.Exists(tempAppDataConfigDir))
-                {
-                    Directory.CreateDirectory(tempAppDataConfigDir);
-                    // Only assign it if creation succeeds. It will throw if it fails.
-                    appDataConfigDirectory = tempAppDataConfigDir;
-                }
-                // Do not catch exceptions here. Let them flow up.
+                // Exceptions are not caught so that they will propagate to the host for display to the user.
+                Directory.CreateDirectory(appDataConfigDirectory);
```

Comment:
CreateDirectory doesn't throw an exception if it already exists. It just returns the directory.
https://msdn.microsoft.com/en-us/library/54a0at6s(v=vs.110).aspx

Now the question becomes... is the check if it exists necessary? It feels like it is slightly faster to check first in scenarios where the directory already exists.